server/src/tests/crypto.rs

```diff
@@ -25,9 +56,9 @@ fn test_get_random_bytes() {
     let mut crypto = FakeCrypto::new();
     let mut buffer = [0; 10];
 
-    crypto.get_random_bytes(&mut buffer);
+    crypto.fill_bytes(&[], &mut buffer).unwrap();
```

Comment:
Note: If we have an internal `FirmwareError` type, the test functions move to `?` try operators, and if we set-up failure modes we can test them more clearly using actual hard error types.

Not too important here, because there are no failure types, but will be important in the future (and maybe worth trying to evoke some failure cases like needing to reseed for better coverage of your traits).

Reply:
So in this particular case we do control the trait. But we control it because no one else created it, for other cases I intend to use already existing traits from rustcrypto ecosystem like this: https://docs.rs/digest/latest/digest/trait.Digest.html


it is no final, but i wanted to have the flexibility that each hardware feature provider has its own error.